lib/mayaUsd/render/MaterialXGenOgsXml/LobePruner.cpp

```diff
@@ -356,39 +444,53 @@ PXR_NS::TfToken LobePrunerImpl::getOptimizedNodeId(const PXR_NS::HdMaterialNode2
     }
 
     if (canOptimize) {
-        ensureLibraryHasOptimizedShader(node.nodeTypeId, flags);
-        return PXR_NS::TfToken(
-            ND_PREFIX + nodeDef->GetFamily().GetString() + "_" + flags + "_surfaceshader");
+        return PXR_NS::TfToken(ensureLibraryHasOptimizedShader(node.nodeTypeId, flags)->getName());
     }
 
     return retVal;
 }
 
-void LobePrunerImpl::ensureLibraryHasOptimizedShader(
+bool LobePrunerImpl::isOptimizedNodeId(const PXR_NS::TfToken& nodeId)
+{
+    return _optimizedNodeIds.count(nodeId) != 0;
+}
+
+mx::NodeDefPtr LobePrunerImpl::ensureLibraryHasOptimizedShader(
```

Comment:
[Thought] Wondering if the name could be improved, now that the method actually returns the node definition. Something like `addOptimizedNodeDef()` would be intuitive but it doesn't capture that the method first checks if it already exists.

Maybe we could just overload `getOptimizedNodeDef()`?

Reply:
Default semantics of "get" is to return noting if not found. Maybe "getOrAdd" would express the intent better?